### PR TITLE
Added http API skeleton server.

### PR DESCRIPTION
**Pull request by socaa**
Part of #11570

### CODE REVIEW

**Comment by piosz** [comment]
@k8s-bot ok to test

**Comment by piosz** [comment]
@googlebot check again!

File: contrib/for-tests/resource-consumer/resource_consumer_handler.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+const (
+	wrongData               = "not enough correct url data"
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/resource_consumer_handler.go]: This error message sounds rather strange

File: contrib/for-tests/resource-consumer/resource_consumer_handler.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	if req.URL.Path == consumeMemAddress {
+		handler.handleConsumeMem(w, query)
+		return
+	}
+	// handle getCurrentStatus
+	if req.URL.Path == getCurrentStatusAddress {
+		handler.handleGetCurrentStatus(w)
+		return
+	}
+	fmt.Fprint(w, wrongData)
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/resource_consumer_handler.go]: Shouldn't we set some HTTP error code here?

File: contrib/for-tests/resource-consumer/resource_consumer_handler.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	empty                   = ""
+)
+
+type ResourceConsumerHandler struct{}
+
+func (handler ResourceConsumerHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	// parsing raw query
+	query, err := url.ParseQuery(req.URL.RawQuery)
+	if err != nil {
+		fmt.Fprint(w, err)
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/resource_consumer_handler.go]: Shouldn't we set some HTTP error code here?

File: contrib/for-tests/resource-consumer/utils.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+*/
+
+package main
+
+import (
+	"github.com/golang/glog"
+)
+
+func ConsumeCPU(milicores int, durationSec int) {
+	glog.Infof("ConsumeCPU milicores: %v, durationSec: %v", milicores, durationSec)
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/utils.go]: This is not implemented yet (I guess you want start a new process here), so, please mark as not implemented.

File: contrib/for-tests/resource-consumer/utils.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func ConsumeMem(megabytes int, durationSec int) {
+	glog.Infof("ConsumeMem megabytes: %v, durationSec: %v", megabytes, durationSec)
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/utils.go]: Add not implemented comment.

File: contrib/for-tests/resource-consumer/utils.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func GetCurrentStatus() {
+	glog.Infof("GetCurrentStatus")
```
Comment (jszczepkowski) [inline contrib/for-tests/resource-consumer/utils.go]: Add not implemented comment.

**Comment by jszczepkowski** [comment]
In general very good.

Please add different error messages, HTTP error codes for failure and comments for unimplemented parts. I think you should also add Dockerfile, as the consumer will run as a container.

File: contrib/for-tests/resource-consumer/utils.go (hunk 4 above)
Comment (socaa) [inline contrib/for-tests/resource-consumer/utils.go]: Done

File: contrib/for-tests/resource-consumer/utils.go (hunk 5 above)
Comment (socaa) [inline contrib/for-tests/resource-consumer/utils.go]: Done

File: contrib/for-tests/resource-consumer/utils.go (hunk 6 above)
Comment (socaa) [inline contrib/for-tests/resource-consumer/utils.go]: Done

**Comment by socaa** [comment]
PTAL

**Comment by jszczepkowski** [comment]
LGTM

**Comment by alex-mohr** [comment]
@k8s-bot test this [contrib/submit-queue: candidate for merging]

**Comment by alex-mohr** [comment]
Automatic merge from SubmitQueue

**Pull request merged**